src/core/evaluator.js

```diff
@@ -4195,8 +4197,13 @@ class PartialEvaluator {
         if (standardFontName) {
           file = await this.fetchStandardFontData(standardFontName);
           properties.isInternalFont = !!file;
-          if (!properties.isInternalFont && this.options.useSystemFonts) {
+          if (
+            !properties.isInternalFont &&
+            this.options.useSystemFonts &&
+            this.localFontCache
```

Comment:
Here, and below, is this new check actually necessary (since we should be able to assume that it's always available when needed)?

Reply:
Good question.
We've some evaluators which haven't the cache because it's useless (or not ??):
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/src/core/document.js#L527 
and I suppose it doesn't make sense to call `getFontSubstitution` in such a case.
So my idea was to be defensive here to avoid any wrong thing (if a font substitution is made in the evaluator but nothing is sent to the FontLoader then the substitution won't work on the canvas).

Reply:
I removed the check and several unit tests are failing.
For example:
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/test/unit/text_layer_spec.js#L24

I think it makes sense to call this function:
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/src/core/evaluator.js#LL4131C13-L4131C13
in the text extraction context: we need to have some font info in order to correctly guess how the glyphs are layed out.
But it's clear that we don't want to load the font in the `FontLoader` in such a context.

Reply:
Doesn't that simply mean that we should include the new cached here as well?
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/src/core/document.js#L510-L537

Reply:
I pretty sure that no fonts are currently loaded on the main-thread during text-extraction, since that uses a custom `handleSetFont` function that doesn't send anything; see
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/src/core/evaluator.js#L2538-L2561
and then compare with the method used during "normal" parsing (note the `translated.send(this.handler);` line)
https://github.com/mozilla/pdf.js/blob/2d2f7b315e6ec8d6b15f80751e4513bd4f273991/src/core/evaluator.js#L982-L1029